connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -308,10 +309,16 @@ void registerRestExtensions(HerderProvider provider, Plugins plugins, ResourceCo
             config.getList(WorkerConfig.REST_EXTENSION_CLASSES_CONFIG),
             config, ConnectRestExtension.class);
 
+        long herderRequestTimeoutMs = ConnectorsResource.REQUEST_TIMEOUT_MS;
+        if (config instanceof DistributedConfig) {
```

Comment:
nit: i don't love `instanceof` solutions. might be better to generalize thee rebalance timeout so support a small/zero value for `StandaloneConfig`.

I wouldn't block merging on this, but this is an anti-pattern, so if there is a cleaner solution, it might be worth exploring.

Reply:
@C0urante, what do you think about doing something like the following instead of the `instanceof` block?

```
        Integer rebalanceTimeoutMs = config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG);
        if (rebalanceTimeoutMs != null) {
            herderRequestTimeoutMs = Math.min(herderRequestTimeoutMs, rebalanceTimeoutMs.longValue());
        }
```
This still uses the `DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG`, but otherwise is simply checking for the existence of that property to determine if `herderRequestTimeoutMs` should be set to a new value.

Reply:
@rhauch sgtm! I've added your code snippet verbatim to the PR